weave/trace_server/interface/base_object_classes/annotation_spec.py

```diff
@@ -41,12 +42,49 @@ class AnnotationSpec(base_object_def.BaseObject):
         ],
     )
 
-    @field_validator("json_schema")
-    def validate_json_schema(cls, v: dict) -> dict:
+    @model_validator(mode="before")
+    @classmethod
+    def preprocess_field_schema(cls, data: dict[str, Any]) -> dict[str, Any]:
+        if "field_schema" not in data:
+            return data
+
+        field_schema = data["field_schema"]
+
+        # Handle Pydantic Field
+        if isinstance(field_schema, FieldInfo):
+            data["field_schema"] = field_schema.json_schema()
+            return data
+
+        # Handle Pydantic model
+        if isinstance(field_schema, type) and issubclass(field_schema, BaseModel):
+            data["field_schema"] = field_schema.model_json_schema()  # type: ignore
+            return data
+
+        return data
+
+    @field_validator("field_schema")
+    def validate_field_schema(cls, schema: dict[str, Any]) -> dict[str, Any]:
+        # Validate the schema
         try:
-            jsonschema.validate(None, v)
+            jsonschema.validate(None, schema)
         except jsonschema.exceptions.SchemaError as e:
             raise e
         except jsonschema.exceptions.ValidationError:
             pass  # we don't care that `None` does not conform
-        return v
+        return schema
+
+    def validate(self, payload: Any) -> bool:
+        """
+        Validates a payload against this annotation spec's schema.
+
+        Args:
+            payload: The data to validate against the schema
+
+        Returns:
+            bool: True if validation succeeds, False otherwise
+        """
```

Comment:
I think including the work "feedback" or in some way tying it to the feedback schema might make sense here, perhaps an example